libvast/src/detail/series_builders.cpp

```diff
@@ -278,6 +277,7 @@ series_builder::series_builder(const vast::type& type, bool are_fields_fixed)
       },
     },
     type);
+  can_change_builder_type_ = not are_fields_fixed;
```

Comment:
In the caf::visit we assign a concrete_type builder to *this. This triggers a default construction of can_change_builder_type_. This would override the false into true and allow the common_type casting.

Reply:
Another alternative would be to make this an argument of the concrete type builder, but I don't have too strong of an opinion on this.